BeeKit/Managers/GoalManager.swift

```diff
@@ -57,17 +57,60 @@ public actor GoalManager {
 
     /// Fetch and return the latest set of goals from the server
     public func refreshGoals() async throws {
-        guard let username = await currentUserManager.username else {
-            try await currentUserManager.signOut()
-            return
+        guard let user = self.currentUserManager.user(context: modelContext) else { return }
+        let goalsUnknown = user.goals.count == 0 || user.updatedAt.timeIntervalSince1970 < 24*60*60
+
+        let userResponse: JSON
+        let goalResponse: JSON
+
+        let deleteMissingGoals: Bool
+        let deletedGoals: JSON
+
+        if (goalsUnknown) {
+            logger.notice("Goals unknown, doing full fetch")
+            // We must fetch the user object first, and then fetch goals afterwards, to guarantee User.updated_at is
+            // a safe timestamp for future fetches without losing data
+            userResponse = JSON(try await requestManager.get(url: "api/v1/users/{username}.json")!)
+            goalResponse = JSON(try await requestManager.get(url: "api/v1/users/{username}/goals.json")!)
+
+            deleteMissingGoals = true
+            deletedGoals = JSON(arrayLiteral: [])
+        } else {
+            // TODO: Use old data to find deleted goals
+            logger.notice("Doing incremental update since \(user.updatedAt, privacy: .public)")
+            userResponse = JSON(try await requestManager.get(url: "api/v1/users/{username}.json", parameters: ["diff_since": user.updatedAt.timeIntervalSince1970 + 1])!)
+            goalResponse = userResponse["goals"]
```

Comment:
`goalResponse` here contains a list of strings, goal names, that have been modified since `diff_since`.

Reply:
Sounds like
1) There's an opportunity for docs to be clearer here
2) It would help to add some unit tests illustrating the shape of goals in this case.